Sources/PostgresNIO/New/Connection State Machine/ConnectionStateMachine.swift

```diff
@@ -842,7 +842,15 @@ struct ConnectionStateMachine {
     // MARK: Consumer
     
     mutating func cancelQueryStream() -> ConnectionAction {
-        preconditionFailure("Unimplemented")
+        guard case .extendedQuery(var queryState, let connectionContext) = self.state, !queryState.isComplete else {
+            preconditionFailure("Tried to cancel stream, without active query")
```

Comment:
```suggestion
            preconditionFailure("Tried to cancel stream without active query")
```
Yes, I'm _this_ pedantic about grammar nits 🙂